variables.tf

```diff
@@ -1,24 +1,31 @@
 variable "hub_virtual_networks" {
   description = "A map of hub virtual networks to create"
   type = map(object({
+    ## As this variable is a map[string]VirtualNetwork, do we need vnet's name here? Would using the key as vnet's name be better?
     name                = string
     address_space       = list(string)
     location            = string
     resource_group_name = string
 
-    bgp_community                   = optional(string, "")
-    ddos_protection_plan            = optional(string, "")
-    dns_servers                     = optional(list(string), [])
+    bgp_community = optional(string)
+    ddos_protection_plan = optional(object({
+      id     = string
+      enable = bool
```

Comment:
For DDoS - I think we should simplify this for the caller, even through the API has these properties they aren't really necessary. I believe we should use an optional string and if it is supplied, then build the correct inputs to the submodule.